pkg/middleware/csrf.go

```diff
@@ -4,10 +4,12 @@ import (
 	"errors"
 	"net/http"
 	"net/url"
-	"strings"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/util"
 )
 
-func CSRF(loginCookieName string) func(http.Handler) http.Handler {
+func CSRF(loginCookieName, defaultPort string, logger log.Logger) func(http.Handler) http.Handler {
```

Comment:
we don't care about the port and never access netAddr.Port so we should be fine to skip all this and pass defaultPort=0 when calling SplitHostPortDefault